src/app/search/loadouts/search-filters/freeform.ts

```diff
@@ -204,6 +305,113 @@ const freeformFilters: FilterDefinition<
       return (loadout) => test(loadout.name) || Boolean(loadout.notes && test(loadout.notes));
     },
   },
+  {
+    keywords: ['light', 'power'],
+    /* t('Filter.PowerKeywords') */
+    description: tl('LoadoutFilter.LoadoutLight'),
+    format: 'range',
+    overload: powerLevelByKeyword,
+    filter: ({ compare, allItems, d2Definitions, selectedLoadoutsStore }) => {
+      if (!d2Definitions || !selectedLoadoutsStore || !allItems) {
+        return () => false;
+      }
+      return (loadout: Loadout) => {
+        if (!isLoadoutCompatibleWithStore(loadout, selectedLoadoutsStore)) {
+          return false;
+        }
+        //
+        const resolvedLoadout = ResolveLoadoutToDimItems(
+          loadout,
+          d2Definitions,
+          allItems,
+          selectedLoadoutsStore,
+        );
+
+        // The UI, at the time of implementing this, only shows the light level using the following rules:
```

Comment:
I think you should probably reuse the code the UI uses:

https://github.com/DestinyItemManager/DIM/blob/7aa6d1299e22ab0c2f8947dfa077fa64c0eedb51/src/app/loadout/LoadoutView.tsx#L217-L233

Which is called like:

```ts
let [items, warnitems] = getItemsFromLoadoutItems(
  itemCreationContext,
  loadoutItems,
  store.id,
  allItems,
  modsByBucket,
);
const loadoutItemsByCategory: Record<BucketSortType, ResolvedLoadoutItem[]> = Object.groupBy(
  items.concat(warnitems),
  (li) => li.item.bucket.sort ?? 'Unknown',
);
const power = loadoutPower(store, loadoutItemsByCategory);
```